R/position-boxdodge.r

```diff
@@ -0,0 +1,116 @@
+#' Position dodge for box plots
```

Comment:
I'd like to give this a name that reflects that it works for any geom with variable widths - i.e. it would also work for `geom_rect()`.